aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -322,6 +323,57 @@ static void rsqrt_kernel(TensorIterator& iter) {
   });
 }
 
+static void cumsum_kernel(Tensor& result, const Tensor& self, int64_t dimension) {
+  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "cumsum_cpu", [&] {
+    using accscalar_t = at::acc_type<scalar_t, false>;
+    const scalar_t * const self_ptr = self.data_ptr<scalar_t>();
+    scalar_t * const res_ptr = result.data_ptr<scalar_t>();
+    int64_t d_size = self.size(dimension);
+    int64_t self_stride = self.stride(dimension);
+    int64_t res_stride = self.stride(dimension);
+    int64_t total = self.numel() / d_size;
+
+    parallel_for(0, total, internal::GRAIN_SIZE, [&](int64_t s, int64_t e) {
```

Comment:
just a question: when do we use parallel_for and when parallel_reduce ?

Reply:
parallel_reduced is used for reduction operation in TensorIterator